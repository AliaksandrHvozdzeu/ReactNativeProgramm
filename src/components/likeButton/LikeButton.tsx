import React, {useRef, useState} from 'react';
import {Animated, View} from 'react-native';
import {styles} from './styles';
import {Icon} from 'react-native-elements';

const LikeButton = () => {
  const [like, setLike] = useState(false);
  const [likeType, setLikeType] = useState('hearto');

  const doLike = () => {
    if (like) {
      fadeOut();
      setLikeType('heart');
      setLike(false);
    } else {
      fadeIn();
      setLikeType('hearto');
      setLike(true);
    }
  };

  const fadeAnim = useRef(new Animated.Value(100)).current;
  const fadeIn = () => {
    Animated.timing(fadeAnim, {
      toValue: 0.3,
      duration: 750,
      useNativeDriver: true,
    }).start();
  };
  const fadeOut = () => {
    Animated.timing(fadeAnim, {
      toValue: 1,
      duration: 1250,
      useNativeDriver: true,
    }).start();
  };

  return (
    <View style={styles.icon}>
      <Animated.View
        style={[
          {
            opacity: fadeAnim,
          },
        ]}>
        <Icon
          style={styles.like}
          type="antdesign"
          name={likeType}
          color={styles.like.color}
          onPress={doLike}
        />
      </Animated.View>
    </View>
  );
};

export default LikeButton;
