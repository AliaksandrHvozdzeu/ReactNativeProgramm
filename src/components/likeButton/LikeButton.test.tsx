import 'react-native';
import React from 'react';
import {test, describe, expect} from '@jest/globals';
import LikeButton from './LikeButton';
import {render} from '@testing-library/react-native';

describe('LikeButton tests', () => {
  test('Renders the LikeButton correctly', () => {
    const tree = render(<LikeButton />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('LikeButton render works correctly', () => {
    render(<LikeButton />);
  });
});
