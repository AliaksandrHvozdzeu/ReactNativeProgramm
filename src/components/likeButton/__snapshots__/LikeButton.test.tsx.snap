// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LikeButton tests Renders the LikeButton correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "display": "flex",
      "flexShrink": 0,
      "height": 25,
      "justifyContent": "flex-end",
      "paddingLeft": 0,
      "position": "absolute",
      "right": 35,
      "top": 34,
      "width": 75,
    }
  }
>
  <View
    style={
      [
        {
          "borderRadius": 3,
          "overflow": "hidden",
        },
        {
          "borderRadius": 3,
        },
        {
          "height": 40,
          "marginHorizontal": 50,
          "marginVertical": 10,
          "width": 50,
        },
        false,
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": false,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "color": "#F7F7F7",
          "flexShrink": 0,
          "opacity": 1,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#008ACE",
            "borderColor": "#2089dc",
            "borderRadius": 3,
            "borderWidth": 0,
            "flexDirection": "row-reverse",
            "justifyContent": "center",
            "padding": 8,
            "position": "absolute",
          }
        }
      >
        <View
          style={
            {
              "marginHorizontal": 5,
              "overflow": "hidden",
            }
          }
        >
          <View>
            <View
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                }
              }
            >
              <Text />
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
